contrib/submit-protocol/src/App/MarketplaceForm.tsx

```diff
@@ -0,0 +1,190 @@
+                iconProps={{iconName: "ConfigurationSolid"}}
+                text="URI"
+                disabled={this.props.disabled}
+              />
+            </div>
+            <Callout
+              role="alertdialog"
+              target={this.uriConfigCalloutBtn.current}
+              onDismiss={this.closeConfigCallout}
+              setInitialFocus={true}
+              hidden={!this.state.uriConfigCallout}
+            >
+              <Stack padding={20}>
+                <TextField
+                  label="Marketplace URI"
+                  prefix="https://api.github.com/repos/"
+                  description="Endpoint should be :owner/:repo/contents/:path?ref=:branch"
+                  onChange={this.setMarketplaceURI}
+                  onBlur={this.setProtocolOptions}
+                />
+              </Stack>
+            </Callout>
+          </Stack>
+        </Stack>
+      </Fabric>
+    );
+  }
+
+  private onConfigCallout = () => {
+    this.setState({uriConfigCallout: !this.state.uriConfigCallout});
+  }
+
+  private closeConfigCallout = () => {
+    this.setState({uriConfigCallout: false});
+  }
+
+  private setMarketplaceURI = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, uri?: string) => {
+    if (uri !== undefined) {
+      this.setState({
+        uri: `https://api.github.com/repos/${uri}`,
+        uriType: MarketplaceUriType.GitHub,
+      });
+    }
+  }
+
+  private setProtocolOptions = async (event: React.FocusEvent<HTMLInputElement>) => {
+    const protocolList = await this.getProtocolList(this.state.uri, this.state.uriType);
+    if (protocolList) {
+      const protocolOptions = [
+        { key: "None", text: "None" },
+        { key: "jobsDivider", text: "-", itemType: DropdownMenuItemType.Divider },
+        { key: "jobsHeader", text: "Protocol Jobs", itemType: DropdownMenuItemType.Header },
+      ];
+      for (const protocolItem of protocolList) {
+        protocolOptions.push({
+          key: protocolItem.name,
+          text: protocolItem.name,
+          data: protocolItem.uri,
+        } as any);
+      }
+      this.setState({ protocolOptions });
+    }
+  }
+
+  private selectProtocol = async (event: React.FormEvent<HTMLDivElement>, option?: IDropdownOption) => {
+    if (option !== undefined) {
+      await this.setState({ selectedProtocol: option });
+      if (option.data !== undefined) {
+        try {
+          const res = await fetch(option.data);
+          const data = await res.text();
+          this.props.onSelectProtocol(data);
+        } catch (err) {
+          alert(`Cannot get ${option.data}`);
+        }
+      }
+    }
+  }
+
+  private getProtocolList = async (uri: string, uriType: MarketplaceUriType) => {
+    if (uriType === MarketplaceUriType.GitHub) {
+      try {
+        const res = await fetch(uri);
+        const data = await res.json();
+        if (Array.isArray(data)) {
+          const protocolList: any[] = [];
```

Comment:
could you annotate the fields you need in this response instead of any?

Reply:
updated